### PR TITLE
Design story fixes

### PR DESCRIPTION
**Pull request by rounak610**
### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update

### Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have added the required tests.

### CODE REVIEW

File: app/workflow_executors/step_executors/impl/next_js_server_test_executor.go [hunk 1]
```diff
@@ -169,16 +168,16 @@ func (e NextJsServerStartTestExecutor) Execute(step steps.ServerStartTestStep) e
 }
 
 func (e NextJsServerStartTestExecutor) AnalyseBuildLogs(buildLogs, directoryPlan, apiKey string) (bool, map[string]interface{}, error) {
-	fmt.Println("Analysing Build Logs", buildLogs)
+	fmt.Println("____Analysing Build Logs____", buildLogs)
```
Comment (luciferlinx101) [inline app/workflow_executors/step_executors/impl/next_js_server_test_executor.go:171]: lets use logger as a standard practice anyways we were moving to it initially use zap logger

File: app/workflow_executors/step_executors/impl/open_ai_next_js_update_code_file_executor.go [hunk 2]
```diff
@@ -223,15 +224,14 @@ func (e *NextJsUpdateCodeFileExecutor) InsertCode(filePath string, lineNumber in
 }
 
 func (e NextJsUpdateCodeFileExecutor) UpdateCodeFile(llmResponse, fileName string, step steps.UpdateCodeFileStep) error {
+	fmt.Printf("____updating file %s_____", fileName)
```
Comment (luciferlinx101) [inline app/workflow_executors/step_executors/impl/open_ai_next_js_update_code_file_executor.go:227]: move to zap logger
also remove unnecessary logs like  `fmt.Println("__payload_____12july", createJobRequest)
`

File: app/tasks/create_execution_job_task.go [hunk 3]
```diff
@@ -174,11 +174,26 @@ func (h *CreateExecutionJobTaskHandler) HandleTask(ctx context.Context, t *asynq
...
+		err = h.activityLogService.CreateActivityLogWithTx(tx, execution.ID, executionStep.ID, "INFO", fmt.Sprintf("Initializing Workspace for automated %s development...", story.Type))
+		if err != nil {
+			tx.Rollback()
+			h.logger.Error("Error creating activity log", zap.Error(err))
+			return err
+		}
+	} else {
+		err = h.activityLogService.CreateActivityLogWithTx(tx, execution.ID, executionStep.ID, "INFO",fmt.Sprintf("Initializing Workspace for automated %s development...", story.Type))
+		if err != nil {
+			tx.Rollback()
```
Comment (luciferlinx101) [inline app/tasks/create_execution_job_task.go:193]: simplify code

`err = h.activityLogService.CreateActivityLogWithTx(tx, execution.ID, executionStep.ID, "INFO", fmt.Sprintf("Initializing Workspace for automated %s development...", story.Type))
		if err != nil {
			tx.Rollback()
			h.logger.Error("Error creating activity log", zap.Error(err))
			return err
		}` this repeated in both if and else

**Review by luciferlinx101 — CHANGES_REQUESTED** [review]

File: app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go [hunk 4]
```diff
@@ -214,11 +213,8 @@ func (openAiCodeGenerator OpenAiNextJsCodeGenerator) Execute(step steps.Generate
 func (openAiCodeGenerator *OpenAiNextJsCodeGenerator) buildFinalInstructionForGeneration(
 	step steps.GenerateCodeStep, storyDir string) (map[string]string, error) {
 	// Initialize the final instruction string
-	finalInstruction, err := openAiCodeGenerator.buildInstructionForFirstExecution(step, storyDir)
-	if err != nil {
-		fmt.Printf("Error building instruction for first execution: %s\n", err.Error())
-		return nil, err
-	}
+	var finalInstruction map[string]string
+	var err error
```
Comment (luciferlinx101) [inline app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go:217]: dont declare `var err error` seperately

File: app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go [hunk 5]
```diff
@@ -535,14 +542,6 @@ func (openAiCodeGenerator *OpenAiNextJsCodeGenerator) GetMessages(systemPrompt s
 					Type: "text",
 					Text: fmt.Sprintf("User Feedback: %s", instruction["feedback"]),
 				},
-				{
```
Comment (luciferlinx101) [inline app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go:538]: why removing it?

**Review by luciferlinx101 — CHANGES_REQUESTED** [review]

File: app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go (hunk 5 above)
Comment (rounak610) [reply app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go:538]: existing code was going twice in single call to llm, first one with prompt where we replace existing code and second in the messages; i verified this with ayush

File: app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go [hunk 6]
```diff
@@ -214,28 +213,30 @@ func (openAiCodeGenerator OpenAiNextJsCodeGenerator) Execute(step steps.Generate
...
 	if step.Retry {
 		fmt.Println("Building instruction on retry limit reached for LLM steps")
+		var err error
 		finalInstruction, err = openAiCodeGenerator.buildInstructionOnRetry(step, storyDir)
 		if err != nil {
 			fmt.Printf("Error building instruction on retry: %s\n", err.Error())
 			return nil, err
 		}
 	} else if step.Execution.ReExecution {
+		var err error
```
Comment (luciferlinx101) [inline app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go:226]: I meant `finalInstruction, err := openAiCodeGenerator.buildInstructionOnReExecutionWithComments(step, storyDir)` should work right?

**Review by luciferlinx101 — CHANGES_REQUESTED** [review]

**Review by luciferlinx101 — APPROVED** [review]

**Pull request merged**